desc/objectives/_geometry.py

```diff
@@ -360,6 +365,150 @@ def compute(self, *args, **kwargs):
+    surrounding surface.
+
+    Parameters
+    ----------
+    surface : Surface
+        Bounding surface to penalize distance to.
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : float, ndarray, optional
+        Target value(s) of the objective.
+        len(target) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+    surface_grid : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate surface geometry at.
+    plasma_grid : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate plasma geometry at.
+    name : str
+        Name of the objective function.
+    """
+
+    _scalar = False
+    _linear = False
+    _units = "(m)"
+    _print_value_fmt = "Minimum plasma-surface distance: {:10.3e} "
+
+    def __init__(
+        self,
+        surface,
+        eq=None,
+        target=0,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        surface_grid=None,
+        plasma_grid=None,
+        name="vessel distance",
+    ):
+        self._surface = surface
+        self._surface_grid = surface_grid
+        self._plasma_grid = plasma_grid
+        super().__init__(
+            eq=eq,
+            target=target,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, eq, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        if self._surface_grid is None:
+            self._surface_grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        if self._plasma_grid is None:
+            self._plasma_grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        if not np.allclose(self._surface_grid.nodes[:, 0], 1):
+            warnings.warn("Surface grid includes off-surface pts, should be rho=1")
+        if not np.allclose(self._plasma_grid.nodes[:, 0], 1):
+            warnings.warn("Plasma grid includes interior points, should be rho=1")
+
+        self._dim_f = self._plasma_grid.num_nodes
```

Comment:
Change so that `dim_f`, `target`, and `weight` correspond to the `surface_grid`, since that is fixed during optimization and more intuitive.